client/components/bulk-select/index.jsx

```diff
@@ -21,33 +19,30 @@ export default React.createClass( {
 	},
 
 	getStateIcon() {
-		if ( this.hasSomeElementsSelected() ) {
+		if ( ! this.hasAllElementsSelected() && this.hasSomeElementsSelected() ) {
 			return <Gridicon className="bulk-select__some-checked-icon" icon="minus-small" size={ 18 }/>;
 		}
 	},
 
 	hasAllElementsSelected() {
-		return this.props.selectedElements && this.props.selectedElements === this.props.totalElements;
```

Comment:
`selectedElements` is declared as a required prop so checking for definition is not needed. same applies in `hasSomeElementsSelected` below